### DISCUSSION TITLE
Save Prisma client instance to a database and retrieve it in my code.

### DISCUSSION THREAD

**Discussion by Easy-Cloud-in** [prisma/prisma#20498] [Q&A]
Is it possible to save a Prisma client instance to a database and retrieve it from my code.

**Comment by janpio** [comment]
Most probably not. A Prisma Client instance has dynamic components like the connection pool, which keeps a list of open connections to the database. These can not be persisted to a database. I am not sure how Prisma Client would handle being loaded again, and then all the connections are gone and have to be recreated.

**Comment by Easy-Cloud-in** [comment]
Thankyou  for the replay. So every logged in user must have their there own Prisma client instance in a multi-tenant app an destroyed on log out. In my use case each client has their own database with the same Prisma schema, so i must maintain a Prisma client pool. Am i correct or there any other way in Prisma for multi-tenant app?

**Reply by janpio** [reply]
No, right now that is the recommended approach indeed.

You would need to keep these Client instances in memory, and manage them that way.

**Reply by Easy-Cloud-in** [reply]
Thank you,  currently, we are maintaining client instances in memory.  
But is there any side effect if the number of clients grows? 
Now we are restricting only 100 clients from logging into the app. We are using PM2  in our server to handle load balancers.

**Reply by janpio** [reply]
Memory and CPU usage will not be optimal most probably. All these Clients maintain their own connection pool and do not share any resources that in theory could be sahred. That comes at a cost.

We have two issues for this.
- One general one: https://github.com/prisma/prisma/issues/2077
- And one that is specific for multi-tenancy via database _schemas_: https://github.com/prisma/prisma/issues/12420

There are third party approaches as well, like https://zenstack.dev/blog/multi-tenant, but I can't say if they are a good idea.

**Reply by Easy-Cloud-in** [reply]
Thank you for your time and for giving me your insight on this issue.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
